File: ABC088B.ml
(* ABC088B - Card Game for Two *)

let solver _ xs =
  let reversed = List.sort (fun x y -> compare y x) xs in
  let f (counter, alice, bob) x =
    if counter mod 2 = 0 then (succ counter, alice + x, bob)
    else (succ counter, alice, bob + x)
  in
  let counter, alice, bob = List.fold_left f (0, 0, 0) reversed in
  alice - bob

let () =
  let n = read_int () in
  read_line ()
  |> Str.split (Str.regexp " ")
  |> List.map int_of_string |> solver n |> string_of_int |> print_endline

File: ABC088B.cpp
// ABC088B - Card Game for Two
#include <iostream>
#include <vector>
#include <algorithm>

long solver(std::vector<long> &A, long N)
{
    std::sort(A.begin(), A.end(), std::greater<long>{});
    auto alice = 0L;
    auto bob = 0L;
    for (auto i = 0L; i < N; i += 2)
    {
        alice += A[i];
        bob += A[i + 1];
    }
    return alice - bob;
}

int main()
{
    long N;
    std::cin >> N;
    std::vector<long> A(100);
    for (auto i = 0; i < N; ++i)
    {
        std::cin >> A[i];
    }
    std::cout << solver(A, N) << std::endl;
    return 0;
}
